src/wp-includes/class-wp-rewrite.php

```diff
@@ -1490,18 +1490,33 @@ public function rewrite_rules() {
 	public function wp_rewrite_rules() {
 		$this->rules = get_option( 'rewrite_rules' );
 		if ( empty( $this->rules ) ) {
-			$this->matches = 'matches';
-			$this->rewrite_rules();
-			if ( ! did_action( 'wp_loaded' ) ) {
-				add_action( 'wp_loaded', array( $this, 'flush_rules' ) );
-				return $this->rules;
-			}
-			update_option( 'rewrite_rules', $this->rules );
+			$this->refresh_rewrite_rules();
 		}
 
 		return $this->rules;
 	}
 
+	/**
+	 * Refreshes the rewrite rules, saving the fresh value to the database.
+	 * If the `wp_loaded` action has not occurred yet, will postpone saving to the database.
+	 *
+	 * @return void
```

Comment:
Per WP core docs standards `@return void` should be omitted.

It would however be good to add a `@since` annotation.
```suggestion
	 * @since 6.4.0
```